cmd/dptp-controller-manager/main.go

```diff
@@ -118,11 +121,33 @@ func newOpts() (*options, error) {
 	fs.Var(&opts.serviceAccountSecretRefresherOptions.ignoreServiceAccounts, "serviceAccountRefresherOptions.ignore-service-account", "The service account to ignore. It must be in namespace/name format (e.G `ci/sync-rover-groups-updater`). Can be passed multiple times.")
 	fs.Var(&opts.imagePusherOptions.imageStreamsRaw, "imagePusherOptions.image-stream", "An imagestream that will be synced. It must be in namespace/name format (e.G `ci/clonerefs`). Can be passed multiple times.")
 	fs.BoolVar(&opts.dryRun, "dry-run", true, "Whether to run the controller-manager with dry-run")
+	fs.StringVar(&opts.releaseRepoGitSyncPath, "release-repo-git-sync-path", "", "Path to release repository dir")
 	if err := fs.Parse(os.Args[1:]); err != nil {
 		logrus.WithError(err).Fatal("could not parse args")
 	}
 
 	var errs []error
+	if opts.releaseRepoGitSyncPath != "" {
+		if opts.ciOperatorconfigPath != "" || opts.stepConfigPath != "" || opts.prowconfig.JobConfigPath != "" || opts.prowconfig.SupplementalProwConfigDirs.String() != "" {
+			errs = append(errs, fmt.Errorf("--release-repo-path is mutually exclusive with --ci-operator-config-path and --step-config-path and --%s and --supplemental-prow-config-dir", opts.prowconfig.JobConfigPathFlagName))
+		} else {
+			if _, err := os.Stat(opts.releaseRepoGitSyncPath); err != nil {
+				if os.IsNotExist(err) {
+					errs = append(errs, fmt.Errorf("--release-repo-path points to a nonexistent directory: %w", err))
+				}
+				errs = append(errs, fmt.Errorf("error getting stat info for --release-repo-path directory: %w", err))
+			}
+
+			opts.ciOperatorconfigPath = filepath.Join(opts.releaseRepoGitSyncPath, config.CiopConfigInRepoPath)
+			opts.stepConfigPath = filepath.Join(opts.releaseRepoGitSyncPath, config.RegistryPath)
+			opts.prowconfig.JobConfigPath = filepath.Join(opts.releaseRepoGitSyncPath, config.JobConfigInRepoPath)
+			opts.prowconfig.ConfigPath = filepath.Join(opts.releaseRepoGitSyncPath, config.ConfigInRepoPath)
```

Comment:
It's not this simple, unfortunately.  This will not handle configuration updates correctly.

The simplest thing to do would be to mimic what the resolver does when it creates the agents here:

https://github.com/openshift/ci-tools/blob/9d0042843f5fafdd8373af7b8769468ec51784c4/cmd/ci-operator-configresolver/main.go#L158-L192

Doing the same when we create the configuration and registry agents here should be enough:

https://github.com/openshift/ci-tools/blob/9d0042843f5fafdd8373af7b8769468ec51784c4/cmd/dptp-controller-manager/main.go#L380
https://github.com/openshift/ci-tools/blob/9d0042843f5fafdd8373af7b8769468ec51784c4/cmd/dptp-controller-manager/main.go#L265

Looking at this list of paths in the diff, it might be tricky to use the same files for the Prow configuration, but I think there would be no harm in continuing to use the CM mounts for them and use the `git-sync` directory for `ci-operator` configuration / registry files.

Reply:
> This will not handle configuration updates correctly.

Not introduced by this PR.
I will make a follow up PR about it as discussed over slack.